### PR TITLE
Implement "--seed NUMBER" flag to seed Math.random() and crypto.getRandomValues()

### LINKED ISSUE #2322: Add --seed for setting Math.random's RNG seed

### PR DESCRIPTION
**Pull request by mtharrison**
Fixes #2322

### CODE REVIEW

**Comment by ry** [comment]
Does this work with `Crypto.getRandomValues()` 

https://github.com/denoland/deno/blob/a115340288d974f141cceb16faac71914402c445/cli/ops.rs#L2213-L2220

Can you add this to `tests/seed_random.js`
```
  const arr = new Int8Array(32);
  console.log(crypto.getRandomValues(arr));
```

File: cli/flags.rs [hunk 1]
```diff
@@ -379,6 +392,19 @@ pub fn parse_flags(matches: &ArgMatches) -> DenoFlags {
     v8_flags.insert(0, "deno".to_string());
     flags.v8_flags = Some(v8_flags);
   }
+  if matches.is_present("seed") {
+    let seed = matches.value_of("seed").unwrap();
+    let v8_seed_flag = format!("--random_seed={}", seed);
```
Comment (bartlomieju) [inline cli/flags.rs]: I should be `--random-seed`

File: cli/flags.rs [hunk 2]
```diff
@@ -1112,4 +1138,19 @@ mod tests {
...
+    assert_eq!(
+      flags,
+      DenoFlags {
+        v8_flags: Some(svec!["deno", "--random_seed=250"]),
+        ..DenoFlags::default()
+      }
+    );
+    assert_eq!(subcommand, DenoSubcommand::Run);
+    assert_eq!(argv, svec!["deno", "script.ts"])
+  }
```
Comment (bartlomieju) [inline cli/flags.rs]: Can you add test case that includes `--v8-flags` as well?

File: cli/flags.rs [hunk 3]
```diff
@@ -145,6 +145,19 @@ To get help on the another subcommands (run in this case):
...
+    ).arg(
+      Arg::with_name("seed")
+        .long("seed")
+        .value_name("NUMBER")
+        .help("Seed Math.random()")
+        .takes_value(true)
+        .validator(|val: String| {
+          match val.parse::<i64>() {
+            Ok(_) => Ok(()),
+            Err(_) => Err("value should be a number".to_string())
```
Comment (bartlomieju) [inline cli/flags.rs]: s/value/Seed/

**Comment by mtharrison** [comment]
@ry this PR doesn't currently affect the values generated by `Crypto.getRandomValues()`, only the values generated by V8 (which I believe the only interface to is `Math.random()`). Would we like to support seeding the Rust RNG too?

File: cli/flags.rs (hunk 3 above)
Comment (mtharrison) [reply cli/flags.rs]: Thanks for the comments, I've addressed them so far.

**Comment by ry** [comment]
>  Would we like to support seeding the Rust RNG too?

Yes - I'd like deno to be completely deterministic some day.

**Comment by mtharrison** [comment]
Thanks 😄

**Pull request merged**